src/java.base/share/classes/java/lang/Boolean.java

```diff
@@ -46,10 +46,18 @@
  * {@code boolean}.
  *
  * <p>This is a <a href="{@docRoot}/java.base/java/lang/doc-files/ValueBased.html">value-based</a>
- * class; programmers should treat instances that are
- * {@linkplain #equals(Object) equal} as interchangeable and should not
- * use instances for synchronization, or unpredictable behavior may
- * occur. For example, in a future release, synchronization may fail.
+ * class; programmers should treat instances that are {@linkplain #equals(Object) equal}
+ * as interchangeable and should not use instances for synchronization, mutexes, or
+ * with {@linkplain java.lang.ref.Reference object references}.
```

Comment:
We have a preview feature disclaimer for References.

Is there a similar one for mutexes? (What exactly do you mean by "mutexes"? Is it distinct from "synchronization"?)

Reply:
Object.wait* and Object.notify are the most obvious APIs to add a disclaimer.
IllegalMonitorStateException is thrown with the class name.  
IdentityException might be preferred and easier for developers to diagnose.